app/views/my_facilities/missed_visits.html.erb

```diff
@@ -1,40 +1,64 @@
+            <p>
+              Patients who were registered during a quarter <%= "(eg: Q#{example_quarter.second}-#{example_quarter.first})" %>, and had no BPs recorded in the following quarter <%= "(Q#{following_quarter.second}-#{following_quarter.first})" %>
+            </p>
           <% when :month %>
             <% example_month = @display_periods.first %>
-            <% following_months = month_short_name_and_year(local_month_start(*example_month) + 1.months) +
-                ', ' + month_short_name_and_year(local_month_start(*example_month) + 2.months) %>
-            <p>Patients who were registered during a month <%= "(eg: #{month_short_name_and_year(local_month_start(*example_month))})" %>,
-              and had no BPs recorded in the following two months <%= "(#{following_months})" %> </p>
-          <% end %>
-        </div>
-
-        <div class="mt-1 mb-3 col-lg-3">
-          <div class="dropdown show">
-            <a class="btn btn-sm btn-outline btn-secondary dropdown-toggle" href="#" role="button" id="dropdownMenuLink" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" style="min-width: 12em;">
-              <%= PeriodSelection::PERIODS[:missed_visits][@selected_period] %>
-            </a>
-
-            <div class="dropdown-menu" aria-labelledby="dropdownMenuLink">
-              <% PeriodSelection::PERIODS[:missed_visits].each do |period, period_name| %>
-                <button type="submit" name="period" value=<%= period %> class="dropdown-item" form="query-filters"><%= period_name %></button>
-              <% end %>
-            </div>
+            <% following_months = month_short_name_and_year(local_month_start(*example_month) + 1.months) + ', ' + month_short_name_and_year(local_month_start(*example_month) + 2.months) %>
+            <p>
+              Patients who were registered during a month <%= "(eg: #{month_short_name_and_year(local_month_start(*example_month))})" %>, and had no BPs recorded in the following two months <%= "(#{following_months})" %>
+            </p>
+        <% end %>
+      </header>
+      <div class="d-flex fw-wrap">
+        <%= render "shared/size_and_zone_filters" %>
+        <div class="dropdown show">
+          <a
+            id="dropdownMenuLink"
+            class="btn btn-sm btn-outline btn-secondary dropdown-toggle" 
+            href="#"
+            role="button"
+            data-toggle="dropdown"
+            aria-haspopup="true"
+            aria-expanded="false"
+          >
+            <%= PeriodSelection::PERIODS[:missed_visits][@selected_period] %>
+          </a>
+          <div class="dropdown-menu" aria-labelledby="dropdownMenuLink">
+            <% PeriodSelection::PERIODS[:missed_visits].each do |period, period_name| %>
+              <button
+                class="dropdown-item filter-by-period <%= "active" if period.to_s == @selected_period.to_s %>"
+                data-value="<%= period %>"
```

Comment:
The period selection JS at the bottom can be removed by tweaking the button to submit the form:
```
<button 
  class="dropdown-item <%= "active" if period.to_s == @selected_period.to_s %>"
  value="<%= period %>"
  type="submit" 
  name="period" 
  form="query-filters">
  <%= period_name %>
</button>
```

Reply:
@prabhanshuguptagit thanks for the feedback! If we remove the JS at the bottom, the selected period is cleared if the user selects an option from a different dropdown. We need the JS to set the right values in the hidden inputs.

Let me know if you know how to address this without JS, as it would also allow us to remove the JS from the block and size filters